packages/admin/cms-admin/src/userPermissions/user/permissions/PermissionDialog.tsx

```diff
@@ -0,0 +1,282 @@
+import { gql, useApolloClient, useQuery } from "@apollo/client";
+import { CancelButton, Field, FinalForm, FinalFormInput, FinalFormSelect, FormSection, SaveButton } from "@comet/admin";
+import { FinalFormDatePicker } from "@comet/admin-date-time";
+import { CircularProgress, Dialog, DialogActions, DialogContent, DialogTitle, InputBaseProps } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import React from "react";
+import { FieldRenderProps, Form } from "react-final-form";
+import { FormattedMessage } from "react-intl";
+
+import { UserPermissionsSettings } from "../../UserPermissionsPage";
+import { camelCaseToHumanReadable } from "../../utils/camelCaseToHumanReadable";
+import {
+    GQLAvailablePermissionsQuery,
+    GQLAvailablePermissionsQueryVariables,
+    GQLCreateUserPermissionMutation,
+    GQLCreateUserPermissionMutationVariables,
+    GQLPermissionQuery,
+    GQLPermissionQueryVariables,
+    GQLUpdateUserPermissionMutation,
+    GQLUpdateUserPermissionMutationVariables,
+    GQLUserPermissionDialogFragment,
+    namedOperations,
+} from "./PermissionDialog.generated";
+
+type ConfigurationProps = InputBaseProps &
+    FieldRenderProps<JSON, HTMLInputElement | HTMLTextAreaElement> & {
+        configurationComponent: () => React.ReactNode;
+    };
+
+const Configuration = ({ configurationComponent, input, disabled }: ConfigurationProps) => {
+    const ConfigurationComponent = ({ values }: { values: JSON }) => {
+        React.useEffect(() => {
+            if (input.value !== values) {
+                input.onChange(values);
+            }
+        }, [values]);
+        return configurationComponent();
```

Comment:
I don't understand how this is supposed to work.

- please provide an example implementation in demo
- or leave out the custom configuration for now and add it later (so we can move faster forward with User Permissions)

Reply:
An example is now available: https://github.com/vivid-planet/comet/pull/1223/commits/3c56a3cf512ff78c3425bf528573cb81b369d18d